### ISSUE TITLE
Incorrect type inferred on descriptor attribute when referenced from 'cls'

### ISSUE THREAD

**Issue by d-tw** [microsoft/pyright#1571]
**Describe the bug**
When accessing a descriptor on the `cls` argument of a `@classmethod`, the descriptor attribute is revealed with the same type as if accessed from an instance of the class.

**To Reproduce**
See code below. The issue occurs on lines 23 and 37 where the type of `field` is inferred as `int`.

**Expected behavior**
The revealed types for `Model.field` and `cls.field` should be identical.

**Screenshots or Code**
```python
# pyright: strict

from __future__ import annotations
from typing import overload, TypeVar, Type, cast

T = TypeVar('T')
M = TypeVar('M', bound='Model')

class Descriptor:
    @overload
    def __get__(self, instance: None, owner: type) -> bool: ...
    @overload
    def __get__(self, instance: T, owner: Type[T]) -> int: ...

    def __get__(self, instance: T, owner: Type[T]) -> int: ...

class Model:
    field: Descriptor = Descriptor()
    
    @classmethod
    def my_method(cls):
        reveal_type(Model.field)
        reveal_type(cls.field)
        reveal_type(cls)
        reveal_type(cast(Type[Model], cls).field)

    @classmethod
    def my_other_method(cls: Type[Model], o: Model):
        reveal_type(Model.field)
        reveal_type(cls.field)
        reveal_type(cls)
        reveal_type(cast(Type[Model], cls).field)

    @classmethod
    def my_generic_method(cls: Type[M], o: M):
        reveal_type(Model.field)
        reveal_type(cls.field)
        reveal_type(cls)
        reveal_type(cast(Type[Model], cls).field)
```

<img width="863" alt="image" src="https://user-images.githubusercontent.com/3317531/109885940-a40df100-7c7f-11eb-918a-bae14a53b8c1.png">

**VS Code extension or command-line**
The issue is present in pyright@1.1.117 (the screenshot is from pylance, which I believe has a different version of pyright, but the output is identical).

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by jakebailey** [comment]
Since you use pylance, I'll warn you that the fix was pushed after I tagged our release, so the bugfix won't make it until next week (sorry 🙁).

**Comment by erictraut** [comment]
Yeah, unfortunately we just missed the cutoff for this week's build.

**Comment by d-tw** [comment]
Great, thanks!

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.118, which I just published. It will also be included in the next release of Pylance.
